### PR TITLE
[Scheduler] fix: EDM schedulers when using the exp sigma schedule.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Otherwise the following would fail:

```python
import torch
from huggingface_hub import hf_hub_download
from diffusers import StableDiffusionXLInstructPix2PixPipeline, EDMEulerScheduler
from diffusers.utils import load_image

edit_file = hf_hub_download(repo_id="stabilityai/cosxl", filename="cosxl_edit.safetensors")

pipe_edit = StableDiffusionXLInstructPix2PixPipeline.from_single_file(edit_file, num_in_channels=8, is_cosxl_edit=True)

pipe_edit.scheduler = EDMEulerScheduler(
    sigma_min=0.002, sigma_max=120.0, sigma_data=1.0, prediction_type="v_prediction", sigma_schedule="exponential"
)
pipe_edit.to("cuda")

prompt = "make it a cloudy day"
image = load_image("https://huggingface.co/datasets/multimodalart/genai-book-images/resolve/main/mountain.png")
pipe_edit(prompt=prompt, image=image).images[0]

```

@apolinario FYI.

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_edm_dpmsolver_multistep.py [hunk 1]
```diff
@@ -295,6 +294,8 @@ def _compute_exponential_sigmas(self, ramp, sigma_min=None, sigma_max=None) -> t
         sigma_min = sigma_min or self.config.sigma_min
         sigma_max = sigma_max or self.config.sigma_max
         sigmas = torch.linspace(math.log(sigma_min), math.log(sigma_max), len(ramp)).exp().flip(0)
+        if not torch.is_tensor(ramp):
```
Comment (yiyixuxu) [inline src/diffusers/schedulers/scheduling_edm_dpmsolver_multistep.py:297]: we make the `ramp` variable here and it is a numpy 
```python
ramp = np.linspace(0, 1, self.num_inference_steps)
```

Comment (sayakpaul) [reply src/diffusers/schedulers/scheduling_edm_dpmsolver_multistep.py:297]: Sorry, I didn't get your point. 

We compute `ramp` using `torch.linspace()` here:
https://github.com/huggingface/diffusers/blob/4d633bfe9a3830dfd50b3f70d8469a575dbaf75a/src/diffusers/schedulers/scheduling_edm_euler.py#L103

And then we compute it with `np.linspace()` here:
https://github.com/huggingface/diffusers/blob/4d633bfe9a3830dfd50b3f70d8469a575dbaf75a/src/diffusers/schedulers/scheduling_edm_euler.py#L213

Comment (yiyixuxu) [reply src/diffusers/schedulers/scheduling_edm_dpmsolver_multistep.py:297]: we make `ramp` we can decide what they are - so there is no need to have the `if` here :)

**Comment by sayakpaul** [comment]
@yiyixuxu LMK if the recent changes work for you.

**Comment by yiyixuxu** [comment]
test still fails though 

let's update it [here](https://github.com/huggingface/diffusers/blob/4d633bfe9a3830dfd50b3f70d8469a575dbaf75a/src/diffusers/schedulers/scheduling_edm_euler.py#L213) to be a torch tensor and remove [this line](https://github.com/huggingface/diffusers/blob/4d633bfe9a3830dfd50b3f70d8469a575dbaf75a/src/diffusers/schedulers/scheduling_edm_euler.py#L219)

so it is consistent between `set_timesteps` and `__init__` (I think it's causing the error) and also, we don't need to convert it to numpy and back to torch
